velox/functions/prestosql/tests/RepeatTest.cpp

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::functions::test;
+
+namespace {
+
+class RepeatTest : public FunctionBaseTest {
+ protected:
+  void testExpression(
+      const std::string& expression,
+      const std::vector<VectorPtr>& input,
+      const VectorPtr& expected) {
+    auto result = evaluate(expression, makeRowVector(input));
+    assertEqualVectors(expected, result);
+  }
+
+  void testExpressionWithError(
+      const std::string& expression,
+      const std::vector<VectorPtr>& input,
+      const std::string& expectedError) {
+    VELOX_ASSERT_THROW(
+        evaluate(expression, makeRowVector(input)), expectedError);
+  }
+};
+} // namespace
+
+TEST_F(RepeatTest, repeat) {
+  const auto elementVector = makeNullableFlatVector<float>(
+      {0.0, -2.0, 3.333333, 4.0004, std::nullopt, 5.12345});
+  const auto countVector =
+      makeNullableFlatVector<int32_t>({1, 2, 3, 0, 4, std::nullopt});
+  VectorPtr expected;
+
+  expected = makeNullableArrayVector<float>({
+      {{0.0}},
+      {{-2.0, -2.0}},
+      {{3.333333, 3.333333, 3.333333}},
+      {{}},
+      {{std::nullopt, std::nullopt, std::nullopt, std::nullopt}},
+      std::nullopt,
+  });
+  testExpression("repeat(C0, C1)", {elementVector, countVector}, expected);
+  testExpression("try(repeat(C0, C1))", {elementVector, countVector}, expected);
+
+  // null constant count argument
+  expected = makeNullableArrayVector<float>({
```

Comment:
nit: use makeNullConstant(REAL(), 6) for brevity

Reply:
The return type of the function signature is `array(T)` so in this case it is actually expecting of `ARRAY(REAL)` instead of `REAL`. Is it possible to make such a struct using `makeNullConstant()`? Tried `makeNullConstant(TypeKind::ARRAY, 6)` but it gave error `not a scalar type!`.

Reply:
Oh. Missed that the vector type is array. You can use this:

`BaseVector::createNullConstant(ARRAY(REAL()), 6, pool());`